System/Random/Internal.hs

```diff
@@ -402,13 +402,16 @@ runPureGenST g action = runST $ runGenStateT g $ action
 
 
 -- | The standard pseudo-random number generator.
-type StdGen = SM.SMGen
+newtype StdGen = StdGen { unStdGen :: SM.SMGen }
+  deriving (Show)
 
 instance RandomGen StdGen where
-  next = SM.nextInt
-  genWord32 = SM.nextWord32
-  genWord64 = SM.nextWord64
-  split = SM.splitSMGen
+  next = second StdGen . SM.nextInt . unStdGen
```

Comment:
This effectively removes `RandomGen SM.SMGen`, which is not what we want.

We either need to add another instance or simply use `GeneralizeNewtypeDeriving` to derive `RandomGen` for `StdGen`

Reply:
Thanks @lehins. I shouldn't have pushed this between meetings - silly mistakes.

I've updated this so we keep `RandomGen SM.SMGen` and use `GeneralisedNewtypeDeriving` for `RandomGen StdGen`.